The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5578959799/jobs/10193863906) [(plain)](https://github.com/rust-lang/rust/commit/dee591427512ac50c6888872c645ff75e02511a0/checks/15107797286/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
  lfs: false
  submodules: false
  set-safe-directory: true
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/should-skip-this.sh
src/ci/scripts/should-skip-this.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-channel.sh
src/ci/scripts/verify-channel.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/collect-cpu-stats.sh
src/ci/scripts/collect-cpu-stats.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-sccache.sh
src/ci/scripts/install-sccache.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/select-xcode.sh
src/ci/scripts/select-xcode.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-clang.sh
src/ci/scripts/install-clang.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-wix.sh
src/ci/scripts/install-wix.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-msys2.sh
src/ci/scripts/install-msys2.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-mingw.sh
src/ci/scripts/install-mingw.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-ninja.sh
src/ci/scripts/install-ninja.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/enable-docker-ipv6.sh
src/ci/scripts/enable-docker-ipv6.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-line-endings.sh
src/ci/scripts/verify-line-endings.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-backported-commits.sh
src/ci/scripts/verify-backported-commits.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-stable-version-number.sh
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/run-build-from-ci.sh
src/ci/scripts/run-build-from-ci.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: dee591427512ac50c6888872c645ff75e02511a0
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
..................iiiiiiiiiiii...................

failures:

---- [codegen] tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs#x86_64 stdout ----

error in revision `x86_64`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-14/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll" "/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC,x86_64" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:158:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:158:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z](\.0)*|%_[0-9]+]}}, {{\{\}\*|ptr|%Type5\*}} align 1 {{%[a-z](\.0)*|%_[0-9]+}}){{.*}}[ "kcfi"(i32 [[TYPE5:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:191:174: note: scanning from here
define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo517h70db37eed51e0624E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !13 {
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:167:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:167:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z](\.0)*|%_[0-9]+]}}, {{\{\}\*|ptr|%Type5\*}} align 1 {{%[a-z](\.0)*|%_[0-9]+}}){{.*}}[ "kcfi"(i32 [[TYPE5:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:205:132: note: scanning from here
define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar517hf303eebedf24b47bE() unnamed_addr #1 !<unknown kind #36> !4 {


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll


-dump-input=help explains the following input dump.
Input was:
<<<<<<
           .
           .
           .
           .
          91: ; Function Attrs: noredzone nounwind nonlazybind 
          92: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo117h93502349188b3e24E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !5 { 
          93: start: 
          94:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
          95:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
          96:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !10, !nonnull !10 
          97:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 -56908610) ] 
          98:  ret void 
          99: } 
         100:  
         101: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar1 
         102: ; Function Attrs: noredzone nounwind nonlazybind 
         103: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar117had3ab10c0b90936dE() unnamed_addr #1 !<unknown kind #36> !4 { 
         104: start: 
         105:  %_1 = alloca %Type1, align 1 
         106:  %b.0 = bitcast %Type1* %_1 to {}* 
         107:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.0 to void ({}*)**), i64 3), align 8, !invariant.load !10, !nonnull !10 
         108:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 -56908610) ] 
         109:  ret void 
         110: } 
         111:  
         112: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
         113: ; Function Attrs: noredzone nounwind nonlazybind 
         114: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !6 { 
         115: start: 
         116:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
         117:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
         118:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !10, !nonnull !10 
         119:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 102973893) ] 
         120:  ret void 
         121: } 
         122:  
         123: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar2 
         124: ; Function Attrs: noredzone nounwind nonlazybind 
         125: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar217h18ebf9f7b2e09ed4E() unnamed_addr #1 !<unknown kind #36> !4 { 
         126: start: 
         127:  %_1 = alloca %Type2, align 1 
         128:  %_3.0 = bitcast %Type2* %_1 to {}* 
         129: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
         130:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to [3 x i64]*)) #2 
         131:  %b.0 = bitcast %Type2* %_1 to {}* 
         132:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to void ({}*)**), i64 3), align 8, !invariant.load !10, !nonnull !10 
         133:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 102973893) ] 
         134:  ret void 
         135: } 
         136:  
         137: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
         138: ; Function Attrs: noredzone nounwind nonlazybind 
         139: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !11 { 
         140: start: 
         141:  %_2 = alloca %Type3, align 1 
         142:  %0 = bitcast [3 x i64]* %a.1 to void ({}*, %Type3*)** 
         143:  %1 = getelementptr inbounds void ({}*, %Type3*)*, void ({}*, %Type3*)** %0, i64 3 
         144:  %2 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** %1, align 8, !invariant.load !10, !nonnull !10 
         145:  call void %2({}* align 1 %a.0, %Type3* align 1 %_2) #2 [ "kcfi"(i32 -1903783964) ] 
         146:  ret void 
         147: } 
         148:  
         149: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar3 
         150: ; Function Attrs: noredzone nounwind nonlazybind 
         151: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar317hd298b3b9b9d428c0E() unnamed_addr #1 !<unknown kind #36> !4 { 
         152: start: 
         153:  %_1 = alloca %Type3, align 1 
         154:  %_3.0 = bitcast %Type3* %_1 to {}* 
         155: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
         156:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to [3 x i64]*)) #2 
         157:  %b.0 = bitcast %Type3* %_1 to {}* 
         158:  %0 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** getelementptr inbounds (void ({}*, %Type3*)*, void ({}*, %Type3*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to void ({}*, %Type3*)**), i64 3), align 8, !invariant.load !10, !nonnull !10 
         159:  call void %0({}* align 1 %b.0, %Type3* align 1 %_1) #2 [ "kcfi"(i32 -1903783964) ] 
         160:  ret void 
         161: } 
         162:  
         163: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo4 
         164: ; Function Attrs: noredzone nounwind nonlazybind 
         165: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo417hb59657770e96a860E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !12 { 
         166: start: 
         167:  %_2 = alloca %Type4, align 1 
         168:  %0 = bitcast [3 x i64]* %a.1 to i32* ({}*, %Type4*)** 
         169:  %1 = getelementptr inbounds i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** %0, i64 3 
         170:  %2 = load i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** %1, align 8, !invariant.load !10, !nonnull !10 
         171:  %_3 = call align 4 i32* %2({}* align 1 %a.0, %Type4* align 1 %_2) #2 [ "kcfi"(i32 1825868617) ] 
         172:  ret void 
         173: } 
         174:  
         175: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar4 
         176: ; Function Attrs: noredzone nounwind nonlazybind 
         177: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar417hf2f55f3071851d4eE() unnamed_addr #1 !<unknown kind #36> !4 { 
         178: start: 
         179:  %_1 = alloca %Type4, align 1 
         180:  %_3.0 = bitcast %Type4* %_1 to {}* 
         181: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo4 
         182:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo417hb59657770e96a860E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.3 to [3 x i64]*)) #2 
         183:  %b.0 = bitcast %Type4* %_1 to {}* 
         184:  %0 = load i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** getelementptr inbounds (i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.3 to i32* ({}*, %Type4*)**), i64 3), align 8, !invariant.load !10, !nonnull !10 
         185:  %_7 = call align 4 i32* %0({}* align 1 %b.0, %Type4* align 1 %_1) #2 [ "kcfi"(i32 1825868617) ] 
         186:  ret void 
         187: } 
         188:  
         189: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo5 
         190: ; Function Attrs: noredzone nounwind nonlazybind 
         191: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo517h70db37eed51e0624E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !13 { 
check:158                                                                                                                                                                                  X~~ error: no match found
check:158     ~~~~~~~
check:158     ~~~~~~~
         193:  %b = alloca [32 x %Type5]*, align 8 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         194:  store [32 x %Type5]* bitcast (<{}>* @alloc_38a9d1c1fccd92e612dd2762da060982 to [32 x %Type5]*), [32 x %Type5]** %b, align 8 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         195:  %_6 = load [32 x %Type5]*, [32 x %Type5]** %b, align 8, !nonnull !10, !align !14, !noundef !10 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         196:  %0 = bitcast [3 x i64]* %a.1 to void ({}*, [32 x %Type5]*)** 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         197:  %1 = getelementptr inbounds void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** %0, i64 3 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         198:  %2 = load void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** %1, align 8, !invariant.load !10, !nonnull !10 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         199:  call void %2({}* align 1 %a.0, [32 x %Type5]* align 1 %_6) #2 [ "kcfi"(i32 391485363) ] 
         200:  ret void 
check:158     ~~~~~~~~~~
         201: } 
check:158     ~~
check:158     ~~
         202:  
check:158     ~
         203: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar5 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         204: ; Function Attrs: noredzone nounwind nonlazybind 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         205: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar517hf303eebedf24b47bE() unnamed_addr #1 !<unknown kind #36> !4 { 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:167                                                                                                                                        X~~ error: no match found
check:167     ~~~~~~~
check:167     ~~~~~~~
         207:  %a = alloca [32 x %Type5]*, align 8 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         208:  store [32 x %Type5]* bitcast (<{}>* @alloc_38a9d1c1fccd92e612dd2762da060982 to [32 x %Type5]*), [32 x %Type5]** %a, align 8 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         209:  %_3.0 = bitcast [32 x %Type5]** %a to {}* 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         210: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo5 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         211:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo517h70db37eed51e0624E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.4 to [3 x i64]*)) #2 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         212:  %b.0 = bitcast [32 x %Type5]** %a to {}* 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         213:  %_10 = load [32 x %Type5]*, [32 x %Type5]** %a, align 8, !nonnull !10, !align !14, !noundef !10 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         214:  %0 = load void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** getelementptr inbounds (void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.4 to void ({}*, [32 x %Type5]*)**), i64 3), align 8, !invariant.load !10, !nonnull !10 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         215:  call void %0({}* align 1 %b.0, [32 x %Type5]* align 1 %_10) #2 [ "kcfi"(i32 391485363) ] 
         216:  ret void 
check:167     ~~~~~~~~~~
         217: } 
check:167     ~~
check:167     ~~
         218:  
check:167     ~
         219: attributes #0 = { inlinehint noredzone nounwind nonlazybind "probe-stack"="__rust_probestack" "target-cpu"="x86-64" "target-features"="-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float" } 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         220: attributes #1 = { noredzone nounwind nonlazybind "probe-stack"="__rust_probestack" "target-cpu"="x86-64" "target-features"="-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float" } 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         221: attributes #2 = { nounwind } 
         222:  
check:167     ~
check:167     ~
         223: !llvm.module.flags = !{!0, !1, !2, !3} 
         224:  
check:167     ~
check:167     ~
         225: !0 = !{i32 7, !"PIC Level", i32 2} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         226: !1 = !{i32 1, !"Code Model", i32 2} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         227: !2 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         228: !3 = !{i32 4, !"kcfi", i32 1} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         229: !4 = !{i32 -1522505972} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         230: !5 = !{i32 -56908610} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~
         231: !6 = !{i32 102973893} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~
         232: !7 = !{i32 -1903783964} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         233: !8 = !{i32 1825868617} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~
         234: !9 = !{i32 391485363} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~
         235: !10 = !{} 
check:167     ~~~~~~~~~~
         236: !11 = !{i32 -1134101214} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~
         237: !12 = !{i32 2137817990} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         238: !13 = !{i32 -227860240} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         239: !14 = !{i64 1} 
>>>>>>
------------------------------------------



---- [codegen] tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs#aarch64 stdout ----

error in revision `aarch64`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-14/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll" "/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC,aarch64" "--dump-input-context" "100"
stdout: none
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:158:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:158:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z](\.0)*|%_[0-9]+]}}, {{\{\}\*|ptr|%Type5\*}} align 1 {{%[a-z](\.0)*|%_[0-9]+}}){{.*}}[ "kcfi"(i32 [[TYPE5:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:191:184: note: scanning from here
define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo517h70db37eed51e0624E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !10 {
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:167:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:167:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z](\.0)*|%_[0-9]+]}}, {{\{\}\*|ptr|%Type5\*}} align 1 {{%[a-z](\.0)*|%_[0-9]+}}){{.*}}[ "kcfi"(i32 [[TYPE5:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:205:142: note: scanning from here
define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar517hf303eebedf24b47bE() unnamed_addr #1 !<unknown kind #36> !1 {


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll


-dump-input=help explains the following input dump.
Input was:
<<<<<<
           .
           .
           .
           .
          91: ; Function Attrs: noredzone nounwind 
          92: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo117h93502349188b3e24E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !2 { 
          93: start: 
          94:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
          95:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
          96:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !7, !nonnull !7 
          97:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 -56908610) ] 
          98:  ret void 
          99: } 
         100:  
         101: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar1 
         102: ; Function Attrs: noredzone nounwind 
         103: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar117had3ab10c0b90936dE() unnamed_addr #1 !<unknown kind #36> !1 { 
         104: start: 
         105:  %_1 = alloca %Type1, align 1 
         106:  %b.0 = bitcast %Type1* %_1 to {}* 
         107:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.0 to void ({}*)**), i64 3), align 8, !invariant.load !7, !nonnull !7 
         108:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 -56908610) ] 
         109:  ret void 
         110: } 
         111:  
         112: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
         113: ; Function Attrs: noredzone nounwind 
         114: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !3 { 
         115: start: 
         116:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
         117:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
         118:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !7, !nonnull !7 
         119:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 102973893) ] 
         120:  ret void 
         121: } 
         122:  
         123: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar2 
         124: ; Function Attrs: noredzone nounwind 
         125: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar217h18ebf9f7b2e09ed4E() unnamed_addr #1 !<unknown kind #36> !1 { 
         126: start: 
         127:  %_1 = alloca %Type2, align 1 
         128:  %_3.0 = bitcast %Type2* %_1 to {}* 
         129: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
         130:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to [3 x i64]*)) #2 
         131:  %b.0 = bitcast %Type2* %_1 to {}* 
         132:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to void ({}*)**), i64 3), align 8, !invariant.load !7, !nonnull !7 
         133:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 102973893) ] 
         134:  ret void 
         135: } 
         136:  
         137: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
         138: ; Function Attrs: noredzone nounwind 
         139: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !8 { 
         140: start: 
         141:  %_2 = alloca %Type3, align 1 
         142:  %0 = bitcast [3 x i64]* %a.1 to void ({}*, %Type3*)** 
         143:  %1 = getelementptr inbounds void ({}*, %Type3*)*, void ({}*, %Type3*)** %0, i64 3 
         144:  %2 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** %1, align 8, !invariant.load !7, !nonnull !7 
         145:  call void %2({}* align 1 %a.0, %Type3* align 1 %_2) #2 [ "kcfi"(i32 -1903783964) ] 
         146:  ret void 
         147: } 
         148:  
         149: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar3 
         150: ; Function Attrs: noredzone nounwind 
         151: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar317hd298b3b9b9d428c0E() unnamed_addr #1 !<unknown kind #36> !1 { 
         152: start: 
         153:  %_1 = alloca %Type3, align 1 
         154:  %_3.0 = bitcast %Type3* %_1 to {}* 
         155: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
         156:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to [3 x i64]*)) #2 
         157:  %b.0 = bitcast %Type3* %_1 to {}* 
         158:  %0 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** getelementptr inbounds (void ({}*, %Type3*)*, void ({}*, %Type3*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to void ({}*, %Type3*)**), i64 3), align 8, !invariant.load !7, !nonnull !7 
         159:  call void %0({}* align 1 %b.0, %Type3* align 1 %_1) #2 [ "kcfi"(i32 -1903783964) ] 
         160:  ret void 
         161: } 
         162:  
         163: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo4 
         164: ; Function Attrs: noredzone nounwind 
         165: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo417hb59657770e96a860E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !9 { 
         166: start: 
         167:  %_2 = alloca %Type4, align 1 
         168:  %0 = bitcast [3 x i64]* %a.1 to i32* ({}*, %Type4*)** 
         169:  %1 = getelementptr inbounds i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** %0, i64 3 
         170:  %2 = load i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** %1, align 8, !invariant.load !7, !nonnull !7 
         171:  %_3 = call align 4 i32* %2({}* align 1 %a.0, %Type4* align 1 %_2) #2 [ "kcfi"(i32 1825868617) ] 
         172:  ret void 
         173: } 
         174:  
         175: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar4 
         176: ; Function Attrs: noredzone nounwind 
         177: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar417hf2f55f3071851d4eE() unnamed_addr #1 !<unknown kind #36> !1 { 
         178: start: 
         179:  %_1 = alloca %Type4, align 1 
         180:  %_3.0 = bitcast %Type4* %_1 to {}* 
         181: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo4 
         182:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo417hb59657770e96a860E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.3 to [3 x i64]*)) #2 
         183:  %b.0 = bitcast %Type4* %_1 to {}* 
         184:  %0 = load i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** getelementptr inbounds (i32* ({}*, %Type4*)*, i32* ({}*, %Type4*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.3 to i32* ({}*, %Type4*)**), i64 3), align 8, !invariant.load !7, !nonnull !7 
         185:  %_7 = call align 4 i32* %0({}* align 1 %b.0, %Type4* align 1 %_1) #2 [ "kcfi"(i32 1825868617) ] 
         186:  ret void 
         187: } 
         188:  
         189: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo5 
         190: ; Function Attrs: noredzone nounwind 
         191: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo517h70db37eed51e0624E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !10 { 
check:158                                                                                                                                                                                            X~~ error: no match found
check:158     ~~~~~~~
check:158     ~~~~~~~
         193:  %b = alloca [32 x %Type5]*, align 8 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         194:  store [32 x %Type5]* bitcast (<{}>* @alloc_38a9d1c1fccd92e612dd2762da060982 to [32 x %Type5]*), [32 x %Type5]** %b, align 8 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         195:  %_6 = load [32 x %Type5]*, [32 x %Type5]** %b, align 8, !nonnull !7, !align !11, !noundef !7 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         196:  %0 = bitcast [3 x i64]* %a.1 to void ({}*, [32 x %Type5]*)** 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         197:  %1 = getelementptr inbounds void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** %0, i64 3 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         198:  %2 = load void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** %1, align 8, !invariant.load !7, !nonnull !7 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         199:  call void %2({}* align 1 %a.0, [32 x %Type5]* align 1 %_6) #2 [ "kcfi"(i32 391485363) ] 
         200:  ret void 
check:158     ~~~~~~~~~~
         201: } 
check:158     ~~
check:158     ~~
         202:  
check:158     ~
         203: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar5 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         204: ; Function Attrs: noredzone nounwind 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         205: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar517hf303eebedf24b47bE() unnamed_addr #1 !<unknown kind #36> !1 { 
check:158     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:167                                                                                                                                                  X~~ error: no match found
check:167     ~~~~~~~
check:167     ~~~~~~~
         207:  %a = alloca [32 x %Type5]*, align 8 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         208:  store [32 x %Type5]* bitcast (<{}>* @alloc_38a9d1c1fccd92e612dd2762da060982 to [32 x %Type5]*), [32 x %Type5]** %a, align 8 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         209:  %_3.0 = bitcast [32 x %Type5]** %a to {}* 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         210: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo5 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         211:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo517h70db37eed51e0624E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.4 to [3 x i64]*)) #2 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         212:  %b.0 = bitcast [32 x %Type5]** %a to {}* 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         213:  %_10 = load [32 x %Type5]*, [32 x %Type5]** %a, align 8, !nonnull !7, !align !11, !noundef !7 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         214:  %0 = load void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** getelementptr inbounds (void ({}*, [32 x %Type5]*)*, void ({}*, [32 x %Type5]*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.4 to void ({}*, [32 x %Type5]*)**), i64 3), align 8, !invariant.load !7, !nonnull !7 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         215:  call void %0({}* align 1 %b.0, [32 x %Type5]* align 1 %_10) #2 [ "kcfi"(i32 391485363) ] 
         216:  ret void 
check:167     ~~~~~~~~~~
         217: } 
check:167     ~~
check:167     ~~
         218:  
check:167     ~
         219: attributes #0 = { inlinehint noredzone nounwind "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         220: attributes #1 = { noredzone nounwind "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         221: attributes #2 = { nounwind } 
         222:  
check:167     ~
check:167     ~
         223: !llvm.module.flags = !{!0} 
         224:  
check:167     ~
check:167     ~
         225: !0 = !{i32 4, !"kcfi", i32 1} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         226: !1 = !{i32 -1522505972} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         227: !2 = !{i32 -56908610} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~
         228: !3 = !{i32 102973893} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~
         229: !4 = !{i32 -1903783964} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         230: !5 = !{i32 1825868617} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~
         231: !6 = !{i32 391485363} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~
         232: !7 = !{} 
check:167     ~~~~~~~~~
         233: !8 = !{i32 -1134101214} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         234: !9 = !{i32 2137817990} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~
         235: !10 = !{i32 -227860240} 
check:167     ~~~~~~~~~~~~~~~~~~~~~~~~
         236: !11 = !{i64 1} 
>>>>>>
------------------------------------------


```

</details>

--- a/compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs
+++ b/compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs
@@ -7,10 +7,10 @@
 ///
 /// For more information about LLVM CFI and cross-language LLVM CFI support for the Rust compiler,
 /// see design document in the tracking issue #89653.
-use core::fmt::Display;
 use rustc_data_structures::base_n;
 use rustc_data_structures::fx::FxHashMap;
 use rustc_hir as hir;
+use rustc_middle::ty::layout::IntegerExt;
 use rustc_middle::ty::{
     self, Const, ExistentialPredicate, FloatTy, FnSig, Instance, IntTy, List, Region, RegionKind,
     TermKind, Ty, TyCtxt, UintTy,
@@ -19,6 +19,7 @@ use rustc_middle::ty::{GenericArg, GenericArgKind, GenericArgsRef};
 use rustc_span::def_id::DefId;
 use rustc_span::sym;
 use rustc_target::abi::call::{Conv, FnAbi};
+use rustc_target::abi::Integer;
 use rustc_target::spec::abi::Abi;
 use std::fmt::Write as _;
 
@@ -93,44 +94,54 @@ fn encode_const<'tcx>(
     dict: &mut FxHashMap<DictKey<'tcx>, usize>,
     options: EncodeTyOptions,
 ) -> String {
-    // L<element-type>[n]<element-value>E as literal argument
+    // L<element-type>[n][<element-value>]E as literal argument
     let mut s = String::from('L');
 
-    // Element type
-    s.push_str(&encode_ty(tcx, c.ty(), dict, options));
+    match c.kind() {
+        // Const parameters
+        ty::ConstKind::Param(..) => {
+            // L<element-type>E as literal argument
 
-    // The only allowed types of const parameters are bool, u8, u16, u32, u64, u128, usize i8, i16,
-    // i32, i64, i128, isize, and char. The bool value false is encoded as 0 and true as 1.
-    fn push_signed_value<T: Display + PartialOrd>(s: &mut String, value: T, zero: T) {
-        if value < zero {
-            s.push('n')
-        };
-        let _ = write!(s, "{value}");
-    }
+            // Element type
+            s.push_str(&encode_ty(tcx, c.ty(), dict, options));
+        }
 
-    fn push_unsigned_value<T: Display>(s: &mut String, value: T) {
-        let _ = write!(s, "{value}");
-    }
+        // Literal arguments
+        ty::ConstKind::Value(..) => {
+            // L<element-type>[n]<element-value>E as literal argument
 
-    if let Some(scalar_int) = c.try_to_scalar_int() {
-        let signed = c.ty().is_signed();
-        match scalar_int.size().bits() {
-            8 if signed => push_signed_value(&mut s, scalar_int.try_to_i8().unwrap(), 0),
-            16 if signed => push_signed_value(&mut s, scalar_int.try_to_i16().unwrap(), 0),
-            32 if signed => push_signed_value(&mut s, scalar_int.try_to_i32().unwrap(), 0),
-            64 if signed => push_signed_value(&mut s, scalar_int.try_to_i64().unwrap(), 0),
-            128 if signed => push_signed_value(&mut s, scalar_int.try_to_i128().unwrap(), 0),
-            8 => push_unsigned_value(&mut s, scalar_int.try_to_u8().unwrap()),
-            16 => push_unsigned_value(&mut s, scalar_int.try_to_u16().unwrap()),
-            32 => push_unsigned_value(&mut s, scalar_int.try_to_u32().unwrap()),
-            64 => push_unsigned_value(&mut s, scalar_int.try_to_u64().unwrap()),
-            128 => push_unsigned_value(&mut s, scalar_int.try_to_u128().unwrap()),
-            _ => {
-                bug!("encode_const: unexpected size `{:?}`", scalar_int.size().bits());
+            // Element type
+            s.push_str(&encode_ty(tcx, c.ty(), dict, options));
+
+            // The only allowed types of const values are bool, u8, u16, u32,
+            // u64, u128, usize i8, i16, i32, i64, i128, isize, and char. The
+            // bool value false is encoded as 0 and true as 1.
+            match c.ty().kind() {
+                ty::Int(ity) => {
+                    let bits = c.eval_bits(tcx, ty::ParamEnv::reveal_all(), c.ty());
+                    let val = Integer::from_int_ty(&tcx, *ity).size().sign_extend(bits) as i128;
+                    if val < 0 {
+                        s.push('n');
+                    }
+                    let _ = write!(s, "{val}");
+                }
+                ty::Uint(_) => {
+                    let val = c.eval_bits(tcx, ty::ParamEnv::reveal_all(), c.ty());
+                    let _ = write!(s, "{val}");
+                }
+                ty::Bool => {
+                    let val = c.try_eval_bool(tcx, ty::ParamEnv::reveal_all()).unwrap();
+                    let _ = write!(s, "{val}");
+                }
+                _ => {
+                    bug!("encode_const: unexpected type `{:?}`", c.ty());
+                }
             }
-        };
-    } else {
-        bug!("encode_const: unexpected type `{:?}`", c.ty());
+        }
+
+        _ => {
+            bug!("encode_const: unexpected kind `{:?}`", c.kind());
+        }
     }
 
     // Close the "L..E" pair

--- a/tests/codegen/sanitizer-cfi-emit-type-metadata-trait-objects.rs
+++ b/tests/codegen/sanitizer-cfi-emit-type-metadata-trait-objects.rs
@@ -53,6 +53,18 @@ impl<'a, T, U> Trait4<'a, U> for T {
     }
 }
 
+pub trait Trait5<T, const N: usize> {
+    fn quux(&self, _: &[T; N]);
+}
+
+#[derive(Copy, Clone)]
+pub struct Type5;
+
+impl<T, U, const N: usize> Trait5<U, N> for T {
+    fn quux(&self, _: &[U; N]) {
+    }
+}
+
 pub fn foo1(a: &dyn Trait1) {
     a.foo();
     // CHECK-LABEL: define{{.*}}4foo1{{.*}}!type !{{[0-9]+}}
@@ -114,7 +126,24 @@ pub fn bar4<'a>() {
     // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE4:[[:print:]]+]]")
 }
 
+pub fn foo5(a: &dyn Trait5<Type5, 32>) {
+    let b = &[Type5; 32];
+    a.quux(&b);
+    // CHECK-LABEL: define{{.*}}4foo5{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE5:[[:print:]]+]]")
+}
+
+pub fn bar5() {
+    let a = &[Type5; 32];
+    foo5(&a);
+    let b = &a as &dyn Trait5<Type5, 32>;
+    b.quux(&a);
+    // CHECK-LABEL: define{{.*}}4bar5{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE5:[[:print:]]+]]")
+}
+
 // CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE1]]"}
 // CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE2]]"}
 // CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE3]]"}
 // CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE4]]"}
+// CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE5]]"}

--- a/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs
+++ b/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs
@@ -77,6 +77,19 @@ impl<'a, T, U> Trait4<'a, U> for T {
     }
 }
 
+pub trait Trait5<T, const N: usize> {
+    fn quux(&self, _: &[T; N]);
+}
+
+pub struct Type5;
+
+impl Copy for Type5 {}
+
+impl<T, U, const N: usize> Trait5<U, N> for T {
+    fn quux(&self, _: &[U; N]) {
+    }
+}
+
 pub fn foo1(a: &dyn Trait1) {
     a.foo();
     // CHECK-LABEL: define{{.*}}4foo1{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
@@ -138,7 +151,24 @@ pub fn bar4<'a>() {
     // CHECK:       call align 4 {{ptr|i32\*}} %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr|%Type4\*}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE4:[[:print:]]+]]) ]
 }
 
+pub fn foo5(a: &dyn Trait5<Type5, 32>) {
+    let b = &[Type5; 32];
+    a.quux(&b);
+    // CHECK-LABEL: define{{.*}}4foo5{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z](\.0)*|%_[0-9]+]}}, {{\{\}\*|ptr|\[32 x %Type5\]\*}} align 1 {{%[a-z](\.0)*|%_[0-9]+}}){{.*}}[ "kcfi"(i32 [[TYPE5:[[:print:]]+]]) ]
+}
+
+pub fn bar5() {
+    let a = &[Type5; 32];
+    foo5(&a);
+    let b = &a as &dyn Trait5<Type5, 32>;
+    b.quux(&a);
+    // CHECK-LABEL: define{{.*}}4bar5{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z](\.0)*|%_[0-9]+]}}, {{\{\}\*|ptr|\[32 x %Type5\]\*}} align 1 {{%[a-z](\.0)*|%_[0-9]+}}){{.*}}[ "kcfi"(i32 [[TYPE5:[[:print:]]+]]) ]
+}
+
 // CHECK: !{{[0-9]+}} = !{i32 [[TYPE1]]}
 // CHECK: !{{[0-9]+}} = !{i32 [[TYPE2]]}
 // CHECK: !{{[0-9]+}} = !{i32 [[TYPE3]]}
 // CHECK: !{{[0-9]+}} = !{i32 [[TYPE4]]}
+// CHECK: !{{[0-9]+}} = !{i32 [[TYPE5]]}

